pkg/agent/run.go

```diff
@@ -116,13 +122,19 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 	}
 
 	if !nodeConfig.NoFlannel {
-		if err := flannel.Run(ctx, nodeConfig, coreClient.CoreV1().Nodes()); err != nil {
+		if err := flannel.Run(ctx, &wg, nodeConfig, coreClient.CoreV1().Nodes()); err != nil {
 			return err
 		}
 	}
 
 	if !nodeConfig.AgentConfig.DisableNPC {
-		if err := netpol.Run(ctx, nodeConfig); err != nil {
+		stopChan := make(chan struct{})
+		healthChan := make(chan *healthcheck.ControllerHeartbeat, healthControllerChannelLength)
+		defer func() {
+			close(stopChan)
+			close(healthChan)
```

Comment:
I was a bit unsure about how to deal with those channnels. Basically, kube-router requires both of them for running all its controllers. It also requires a `WaitGroup`

In the other parts of k3s agent, we were not using any channels or WaitGroups, we were just relying in a `Context`. I was wondering - should we inroduce channels and WaitGroups everywhere? Modify kube-router to not require channels?

So far I decided to:
* keep the healtcheck server of kube-router, of course enable only monitoring of netpol controller
* keep the channels only for kube-router
* use the WaitGroup also for the flannel goroutime, because why not :man_shrugging: after all, the purpose of WaitGroups is to wait for all goroutines

WDYT?

Reply:
I think stopChan should come from `ctx.Done()` so that it's plumbed through?  Not sure about the healthChan though, as discussed during standup.